ExampleMod/Common/GlobalProjectiles/ProjectileWithGrowingDamage.cs

```diff
@@ -0,0 +1,41 @@
+﻿using ExampleMod.Common.GlobalItems;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.GlobalProjectiles
+{
+	public class ProjectileWithGrowingDamage : GlobalProjectile
+	{
+		//Related to GlobalItem: WeaponWithGrowingDamage
+
+		private Item sourceItem;
+
+		public override bool InstancePerEntity => true;
+		public override void OnSpawn(Projectile projectile, IEntitySource source) {
+			if (source is EntitySource_ItemUse itemSource) {
+				sourceItem = itemSource.Item;
```

Comment:
Unfortunately you cannot do this, unless you clone it (which I would not recommend).

Because `Item` instances are reused in Terraria (via `SetDefaults`) the item could change into air, be swapped with a different item, be replaced with one you picked up, etc before the projectile lands.

Reply:
I added in the type checking.  Do you think that will be good enough?  It would just fail to gain xp if the item was changed, but the vast majority of the time, it would work.  This is the way I handle it in my mod and haven't noticed issues with it.  (I hadn't considered the game re-using the item instance after setting it as a source item, so I should add some additional checking there.)  It is missing the search method I use if the item has changed inventory slots.  Do you think I should add that in?

Reply:
I wouldn't worry about searching for the item if it's moved to a new slot, but I don't want to show storing the `Item` from `OnSpawn` in `ExampleMod` because even if you know how to safely guard against the item instance changing, others will not

Reply:
updated